CDCgov/prime-reportstream
# Sample message #2 HCI mapping updates

This PR makes another batch of adjustments to mappings based on HCI's second sample message.

Test Steps:
1. Run integration tests
2. Use CLI commands to run manual conversion on the HCI sample message from Sharepoint
3. Run `./prime hl7data -c src/testIntegration/resources/datatests/HL7_to_FHIR_to_HL7/hci-expected.hl7 -s src/testIntegration/resources/datatests/HL7_to_FHIR_to_HL7/hci.hl7`

## Changes
- Remove default values for MSH.17/MSH.18 (this required updates to several test files, waiting to hear back from Victor about whether or not we need a receiver transform there for CA)
- Add a receiver transform to maintain original behavior for MSH.18 for SR->CA (MSH.17 was already being sent by SR)
- Add extensions for XCN->Practitioner.Name to support and preserve the different potential suffix types
- Update ExtensionAssigningAuthority to preserve HD.2 even when HD.3 is not set (this addresses an issue with PL.4, but as a short-term solution, does not fix the mapping for PL.4 altogether)


## Checklist

### Testing

## Linked Issues
- Addresses #11317
